### PR TITLE
Docs with Collaborative Editing enabled are only editable on test server (LessestWrong)

### PR DESCRIPTION
**Pull request by darkruby501**
This PR makes it so that if a document has been edited in collaborative mode, it can only be edited on LessestWrong/test server. This lets us start testing with some users and have them use LessestWrong without the issues if they tried to also edit on LessWrong.

LessWrong 2022-01-27 16-15-33

### CODE REVIEW

File: packages/lesswrong/lib/collections/posts/custom_fields.ts [hunk 1]
```diff
@@ -920,6 +921,18 @@ addFieldsDict(Posts, {
     }
   },
 
+  sharingSettings: {
```
Comment (jimrandomh) [inline packages/lesswrong/lib/collections/posts/custom_fields.ts:924]: This needs a `hidden: true`; as it is now, it adds a visible "Sharing Settings" field to the new-post form, above the title.

**Comment by jimrandomh** [comment]
Fixed one significant issue (this added a non-working prominent field to the new-post and edit-post forms). With that fixed I think it's mergeable.

**Review by jimrandomh — APPROVED** [review]

File: packages/lesswrong/components/posts/PostsEditForm.tsx [hunk 2]
```diff
@@ -45,7 +46,28 @@ const PostsEditForm = ({ documentId, eventForm, classes }: {
     collectionName: "Posts",
     fragmentName: 'SuggestAlignmentPost',
   })
-
+  
+  function isCollaborative(post): boolean {
+    if (!post) return false;
+    if (!post._id) return false;
+    // if (fieldName !== "contents") return false;
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/PostsEditForm.tsx:53]:
```suggestion
```

File: packages/lesswrong/lib/instanceSettings.ts [hunk 3]
```diff
@@ -127,3 +127,6 @@ export const sentryReleaseSetting = new PublicInstanceSetting<string|null>('sent
 export const siteUrlSetting = new PublicInstanceSetting<string>('siteUrl', getAbsoluteUrl(), "optional")
 
 // Stripe setting
+
+//Test vs Production Setting
+export const testServerSetting = new PublicInstanceSetting<boolean>("testServer", false, "warning")
```
Comment (jpaddison3) [inline packages/lesswrong/lib/instanceSettings.ts:132]: Agree with this move.

**Review by jpaddison3 — APPROVED** [review]
Agree with Jim about the bug, but otherwise nice work. Thanks for entertaining an alternate hypothesis about how to prevent the problems you were running into.

**Comment by jpaddison3** [comment]
Super mad at myself for not catching that this applied to the EA Forum. C'mon JP.

**Pull request merged**